yjit/src/core.rs

```diff
@@ -1529,7 +1527,8 @@ fn branch_stub_hit_body(branch_ptr: *const c_void, target_idx: u32, ec: EcPtr) -
     let branch_size_on_entry = branch.code_size();
 
     let target_idx: usize = target_idx.as_usize();
-    let target = branch.targets[target_idx];
+    assert!(branch.targets[target_idx].is_some());
+    let target = branch.targets[target_idx].unwrap();
```

Comment:
```suggestion
    let target = branch.targets[target_idx].unwrap();
```
just a nit. `.unwrap()` checks and panics too, just with a different message. Could use `expect` here to say that we expect branches from running code to be fully constructed.